### PR TITLE
Feature/elliott/bp example

### PR DESCRIPTION
**Pull request by nselliott**
# Summary

- This PR is a modification of some example code.
- It does the following:
  - Modifies sidre_createdatastore.cpp by simplifying the process of setting up a blueprint
    for a simple mesh.
  - Makes some of the examples in this file use Views that see pre-existing Buffers as Buffers, while
    other examples have Views see the pre-existing data as described external pointers, thus providing
    test coverage for both use cases.
  - Addresses issue #45 
  - Documentation in sidre_conduit.rst uses code snippets from this file, so the documentation text has
    been edited to be consistent with these code changes.

### CODE REVIEW

**Comment by nselliott** [comment]
@agcapps I've edited the documentation but left the figures untouched.  The key change is that there is only one DataStore, so the `tinymesh` subtree should now be a child of `ds.root` parallel to `nodes` and `fields`.

File: src/axom/sidre/docs/sphinx/sidre_conduit.rst [hunk 1]
```diff
@@ -61,9 +60,9 @@ First build top-level groups required by the Blueprint.
    :end-before: _blueprint_restructure_toplevel_end
    :language: C++
 
-Add the node coordinates.  The Groups in ``cds`` will use the Buffers within ``ds``
-as external pointers that have a description: while Sidre and Conduit will use the
-array type and shape information, ``cds`` will not deallocate the memory.
+Add the node coordinates.  The Views in ``tinymesh`` will point to the Buffers within ``nodes``
```
Comment (agcapps) [inline src/axom/sidre/docs/sphinx/sidre_conduit.rst]: Maybe rephrase to something like "The Views under ``tinymesh`` will point to the same Buffers as the Views under ``nodes`` and ``fields`` so that ``tinymesh`` can use the data ...".  The intent being to emphasize that the Buffers are shared, in the same Dataset, but not contained or owned by any Groups.

**Comment by nselliott** [comment]
The figures look good, thanks @agcapps
